packages/realm-web/src/App.ts

```diff
@@ -247,6 +258,38 @@ export class App<
         return Object.freeze([...activeUsers, ...loggedOutUsers]);
     }
 
+    /**
+     * @returns A promise of the app URL, with the app location resolved.
+     */
+    public get locationUrl(): Promise<string> {
+        if (!this._locationUrl) {
+            const path = routes.api().app(this.id).location().path;
+            this._locationUrl = this.fetcher
+                .fetchJSON({
+                    method: "GET",
+                    url: this.baseUrl + path,
+                    tokenType: "none",
+                })
+                .then(
+                    ({ hostname }) => {
+                        if (typeof hostname !== "string") {
+                            throw new Error(
+                                "Expected response to contain a 'hostname'",
+                            );
+                        } else {
+                            return hostname;
+                        }
+                    },
+                    err => {
```

Comment:
Personal opinion: I prefer the rejection-handler to be in it's own `catch` chain. (If you disagree, just resolve this one)

Reply:
I guess it depends on what I want to achieve. The semantics are not exactly equivalent.
But ... In this case I probably do want to reset the location if the response doesn't contain a hostname.

Reply:
The current rejection-handler (2nd arg in the `then`) will also catch a rejection/throw from `fetchJSON`, and reset `_locationUrl`, if that's what you are trying to avoid?

```js
Promise.reject(new Error('NOPE'))
    .then(() => { console.log('THEN') }, (err) => { console.log('NESTED:', err) });
// outputs: NESTED: Error: NOPE
```

Just FYI.

Reply:
Yea - I get that 🙂
What I wanted to get across is that I should probably use catch here to ensure the `_locationUrl` is reset *also* for the case where the response is missing the hostname. That case would not had hit the 2nd argument callback in the original code.